tests/unit/s2n_client_server_name_extension_test.c

```diff
@@ -92,6 +94,31 @@ int main(int argc, char **argv)
         EXPECT_SUCCESS(s2n_client_server_name_extension.recv(server_conn, &stuffer));
         EXPECT_STRING_EQUAL(server_conn->server_name, test_server_name);
 
+        EXPECT_EQUAL(s2n_stuffer_data_available(&stuffer), 0);
+
+        EXPECT_SUCCESS(s2n_stuffer_free(&stuffer));
+        EXPECT_SUCCESS(s2n_connection_free(client_conn));
+        EXPECT_SUCCESS(s2n_connection_free(server_conn));
+    }
+
+    /* recv - server name already set */
+    {
+        struct s2n_connection *client_conn;
+        EXPECT_NOT_NULL(client_conn = s2n_connection_new(S2N_CLIENT));
+
+        struct s2n_connection *server_conn;
+        EXPECT_NOT_NULL(server_conn = s2n_connection_new(S2N_CLIENT));
+
+        struct s2n_stuffer stuffer;
+        EXPECT_SUCCESS(s2n_stuffer_growable_alloc(&stuffer, 0));
+
+        EXPECT_SUCCESS(s2n_set_server_name(client_conn, "DIFFERENT SERVER NAME"));
+        EXPECT_SUCCESS(s2n_client_server_name_extension.send(client_conn, &stuffer));
+
+        EXPECT_SUCCESS(s2n_set_server_name(server_conn, test_server_name));
+        EXPECT_SUCCESS(s2n_client_server_name_extension.recv(server_conn, &stuffer));
+        EXPECT_STRING_EQUAL(server_conn->server_name, test_server_name);
```

Comment:
If server_name is already set, we are doing nothing. Which means this should be 
`EXPECT_STRING_EQUAL(server_conn->server_name, "DIFFERENT SERVER NAME");` right?

Reply:
Not quite. "DIFFERENT SERVER NAME" is the name being sent by the client. The server's server_name is set to test_server_name. The server ignores "DIFFERENT SERVER NAME" and keeps test_server_name.